evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -100,16 +100,19 @@ def transform(self, X, y=None):
 
         X_null_dropped = X.copy()
         X_null_dropped.drop(self._all_null_cols, inplace=True, axis=1, errors='ignore')
-        X_null_dropped.reset_index(inplace=True, drop=True)
         if X_null_dropped.empty:
             return X_null_dropped
 
         if self._numeric_cols is not None and len(self._numeric_cols) > 0:
             X_numeric = X_null_dropped[self._numeric_cols]
-            X_null_dropped[X_numeric.columns] = self._numeric_imputer.transform(X_numeric)
+            imputed = self._numeric_imputer.transform(X_numeric)
+            imputed.index = X_null_dropped.index
```

Comment:
Is there ever a case where `imputed.shape != X_null_dropped.shape`? I.e., could `self._numeric_imputer.transform(X_numeric)` ever result in dropped rows?

I believe the answer is, to the best of our knowledge, no. Correct?

Reply:
Yes, I think you're right!